src/microbe_stage/SpawnSystem.cs

```diff
@@ -197,31 +205,16 @@ private void SpawnEntities(Vector3 playerPosition, int existing)
                         previousSquaredDistance > spawnType.SpawnRadiusSqr)
                     {
                         // Second condition passed. Spawn the entity.
-                        var entities = spawnType.Spawn(worldRoot,
+                        var spawnedEntities = spawnType.Spawn(worldRoot,
                             playerPosition + displacement);
 
-                        // Add the entity to the spawned group and add the despawn radius
-                        if (entities != null)
-                        {
-                            foreach (var entity in entities)
-                            {
-                                // TODO: I don't understand why the same
-                                // value is used for spawning and
-                                // despawning, but apparently it works
-                                // just fine
-                                entity.DespawnRadiusSqr = spawnType.SpawnRadiusSqr;
-
-                                entity.SpawnedNode.AddToGroup(Constants.SPAWNED_GROUP);
```

Comment:
I really like the approach of the spawn system being responsible for adding the Node to the spawned group.